hilti/toolchain/src/compiler/unit.cc

```diff
@@ -316,7 +316,8 @@ Result<Nothing> Unit::compile() {
 Result<Nothing> Unit::codegen() {
     auto& module = imported(_id);
 
-    _dumpASTs(logging::debug::AstCodegen, "AST for codegen");
+    // We do not include scopes for the final AST as we do not keep their references up to date.
+    _dumpASTs(logging::debug::AstCodegen, "AST for codegen", 0, false);
```

Comment:
I'm wondering if instead we should clear out the scopes, then there's no risk of anybody accidentally trying to do something with that stale information (). Could make `Unit::_resetNodes` a public function for that and call it just before codegen. (Then we also don't need the changes to `dumpAST`).

Reply:
I tried that, but simply calling `_resetNodes` seems to do too much. I'd prefer to leave this as is for now.

Reply:
I looked into this again and was able to just reset the scopes. This needed a small fix in codegen where we were retrieving a decl which could get optimized away, but never used it.